provider/marathon/builder_test.go

```diff
@@ -0,0 +1,133 @@
+package marathon
+
+import "github.com/gambol99/go-marathon"
+
+// Functions related to building applications.
+
+func application(ops ...func(*marathon.Application)) marathon.Application {
+	app := marathon.Application{}
+	app.EmptyLabels()
+
+	for _, op := range ops {
+		op(&app)
+	}
+
+	return app
+}
+
+func appID(name string) func(*marathon.Application) {
+	return func(app *marathon.Application) {
+		app.Name(name)
+	}
+}
+
+func appPorts(ports ...int) func(*marathon.Application) {
+	return func(app *marathon.Application) {
+		app.Ports = append(app.Ports, ports...)
+	}
+}
+
+func labels(l map[string]string) func(*marathon.Application) {
```

Comment:
I was leaning towards such an implementation too at some point, so I made the change. 👍